google/cloud/bigquery/job/query.py

```diff
@@ -1489,6 +1505,101 @@ def to_dataframe(
             progress_bar_type=progress_bar_type,
             create_bqstorage_client=create_bqstorage_client,
             date_as_object=date_as_object,
+            geography_as_object=geography_as_object,
+        )
+
+    # If changing the signature of this method, make sure to apply the same
+    # changes to table.RowIterator.to_dataframe(), except for the max_results parameter
+    # that should only exist here in the QueryJob method.
+    def to_geodataframe(
+        self,
+        bqstorage_client: "bigquery_storage.BigQueryReadClient" = None,
+        dtypes: Dict[str, Any] = None,
+        progress_bar_type: str = None,
+        create_bqstorage_client: bool = True,
+        date_as_object: bool = True,
```

Comment:
Noted. 

Given that, do you want to change the way we're indicating conversion to shapley geometry?

Reply:
I'm not as concerned about having a flag for geometry as object. I don't think Shapely suffers the same dtype weirdness that DATE columns do, where there's a risk of an exception if you make the wrong choice.

In date, because the supported range of dates in pandas is smaller than the supported range of dates in BigQuery, we can sometimes have to fallback to object. I'd rather not force the user to know the range of results they're expecting in order to keep `to_dataframe` from crashing, which is why I dislike `date_as_object` from an API design perspective.

Reply:
Cool